pkg/controller/utils/utils.go

```diff
@@ -280,6 +281,28 @@ func ValidateCertPair(client client.Client, namespace, certPairSecretName, keyNa
 	return secret, nil
 }
 
+func GetK8sServiceEndPoint(client client.Client) (*k8sapi.ServiceEndpoint, error) {
```

Comment:
We try and add a doc string to every exported type to describe what it is used for succinctly.